### PR TITLE
Upgrade to Keeper 0.8

### LINKED ISSUE #82: get `conditionKeys` from keeper-contracts rather than DDO.service
The DDO isn't the source of truth for the `conditionKeys`, but `keeper-contracts.SLA.serviceAgreementTemplate` is.

This leads to interesting behavior when re-computing the `conditionKeys`. For example one could have a faulty DDO, or one created before a contract migration.

see: https://github.com/oceanprotocol/squid-js/blob/develop/src/ocean/ServiceAgreements/ServiceAgreement.ts#L74

This also applies to `squid-py`.

Another way to avoid is to exclude the `conditionKeys` from the hash. Which might lead to problems with mapping `conditionValues`

### LINKED ISSUE #112: RegisterServiceAgreementTemplates.ts throws error
## Current Behavior

RegisterServiceAgreementTemplates.ts throws error on public key comparison where public keys are the same but with different upper-lower cases.

## Step to Reproduce the problem

Running "npm run run src/examples/RegisterServiceAgreementTemplates.ts"

## Specifications

* Squid-js Version: 0.2.7

## Failure Logs

```
Jernejs-MacBook-Pro:squid-js jernejpregelj$ npm run run src/examples/RegisterServiceAgreementTemplates.ts

> @oceanprotocol/squid@0.2.7 run /Users/jernejpregelj/oceanprotocol/squid-js
> ts-node "src/examples/RegisterServiceAgreementTemplates.ts"

Loaded OceanMarket from spree
Loaded OceanAuth from spree
Loaded OceanToken from spree
Loaded ServiceAgreement from spree
Loaded AccessConditions from spree
Loaded PaymentConditions from spree
Loaded DIDRegistry from spree
Ocean {
  keeper:
   Keeper {
     market:
      OceanMarket { contract: [Contract], contractName: 'OceanMarket' },
     auth:
      OceanAuth { contract: [Contract], contractName: 'OceanAuth' },
     token:
      OceanToken { contract: [Contract], contractName: 'OceanToken' },
     serviceAgreement:
      ServiceAgreement { contract: [Contract], contractName: 'ServiceAgreement' },
     accessConditions:
      AccessConditions { contract: [Contract], contractName: 'AccessConditions' },
     paymentConditions:
      PaymentConditions { contract: [Contract], contractName: 'PaymentConditions' },
     didRegistry:
      DIDRegistry { contract: [Contract], contractName: 'DIDRegistry' } } }
error Error: Template owner missmatch on Access! Should be "0x00bd138abd70e2f00903268f3db08f2d25677c9e" but is 0x00Bd138aBD70e2F00903268F3Db08f2D25677C9e
    at ServiceAgreementTemplate.<anonymous> (/Users/jernejpregelj/oceanprotocol/squid-js/src/ocean/ServiceAgreements/ServiceAgreementTemplate.ts:80:19)
    at step (/Users/jernejpregelj/oceanprotocol/squid-js/src/ocean/ServiceAgreements/ServiceAgreementTemplate.ts:45:23)
    at Object.next (/Users/jernejpregelj/oceanprotocol/squid-js/src/ocean/ServiceAgreements/ServiceAgreementTemplate.ts:26:53)
    at fulfilled (/Users/jernejpregelj/oceanprotocol/squid-js/src/ocean/ServiceAgreements/ServiceAgreementTemplate.ts:17:58)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)```

### LINKED ISSUE #124: Implement missing methods of new Ocean API.
Comment from here: https://github.com/oceanprotocol/squid-js/pull/113#issuecomment-457587296
API specification: https://github.com/oceanprotocol/dev-ocean/blob/squid/trilobite-interface/doc/architecture/squid.md

### LINKED ISSUE #125: Update new consuming flow to download a binary instead of return a URL.
Brizo is going to download a binary instead of returning a new URL to consume data.
Related to https://github.com/oceanprotocol/brizo/pull/81

### PR DESCRIPTION
**Pull request by Pedro-vk**
## Description

Upgrade to Keeper 0.8

## Is this PR related with an open issue?

Related to Issue https://github.com/oceanprotocol/ocean/issues/283
Closes #125
Closes #124 
Closes #112
Closes #82

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [x] Follows the code style of this project.
- [x] Tests Cover Changes
- [x] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,8 +21,9 @@ before_script:
   - ganache-cli --port 18545 > ganache-cli.log &
   - git clone https://github.com/oceanprotocol/barge
   - cd barge
-  - export KEEPER_VERSION=v0.6.12
-  - bash -x start_ocean.sh --latest --no-brizo --no-pleuston --local-spree-node 2>&1 > start_ocean.log &
+  - export KEEPER_VERSION=v0.8.5
+  - export KEEPER_OWNER_ROLE_ADDRESS="0xe2DD09d719Da89e5a3D0F2549c7E24566e947260"
+  - bash -x start_ocean.sh --latest --no-pleuston --local-spree-node --no-acl-contract 2>&1 > start_ocean.log &
```
Comment (ttmc) [inline .travis.yml]: Is it still necessary to use the `--no-acl-contract` option when running Barge? I thought that was just temporary to get around some short-term issues.

Comment (Pedro-vk) [reply .travis.yml]: Is just for testing purposes, so it's not a big problem. In any case, I'll ask for alternatives, thanks :+1:

Comment (aaitor) [reply .travis.yml]: yep, this should be tested in travis without that flag

File: src/ocean/OceanAssets.ts [hunk 2]
```diff
@@ -256,20 +236,22 @@ export default class OceanAssets {
 
         const ddo = await this.resolve(did)
 
+        const keeper = await Keeper.getInstance()
+        const templateName = ddo.findServiceByType("Access").serviceAgreementTemplate.contractName
+        const template = await keeper.getTemplateByName(templateName)
```
Comment (jernejpregelj) [inline src/ocean/OceanAssets.ts]: Not asyc function so await not needed

File: src/ocean/OceanAssets.ts [hunk 3]
```diff
@@ -256,20 +236,22 @@ export default class OceanAssets {
...
+        const accessCondition = await keeper.conditions.accessSecretStoreCondition
```
Comment (jernejpregelj) [inline src/ocean/OceanAssets.ts]: Not asyc function so await not needed

File: src/ocean/OceanAgreementsConditions.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        amount: number,
+        did: string,
+        consumer: string,
+        publisher: string,
+        from: Account = new Account(),
+    ) {
+        const {escrowReward, accessSecretStoreCondition, lockRewardCondition} = this.keeper.conditions
+
+        const conditionIdAccess = await accessSecretStoreCondition.generateIdHash(agreementId, did, consumer)
+        const conditionIdLock = await lockRewardCondition.generateIdHash(agreementId, await escrowReward.getAddress(), amount)
```
Comment (jernejpregelj) [inline src/ocean/OceanAgreementsConditions.ts]: escrowReward.getAddress() not async call

**Comment by jernejpregelj** [comment]
Are /src/examples still needed? Note that that code is packaged in npm package and could make package bigger.

File: src/ocean/OceanAgreementsConditions.ts (hunk 4 above)
Comment (Pedro-vk) [reply src/ocean/OceanAgreementsConditions.ts]: Really good catches! I'm pushing the fixes :) Thanks!

**Comment by Pedro-vk** [comment]
> Are /src/examples still needed? Note that that code is packaged in npm package and could make package bigger.

@jernejpregelj They are not used on the build and not added to the NPM artifact.
I've in my todo list move these examples to the root folder, improve how they are shown and fix them in general, because they are not going to work.
But I'll do it in other PR, there is to much work here.
Thanks :)

**Comment by Pedro-vk** [comment]
> Wrote reported errors regarding async/await. There are some unused imports in ocean/Ocean.ts but that is it from my side.

They seem unused but are necessary for the typings of the deprecated methods.

File: .travis.yml (hunk 1 above)
Comment (Pedro-vk) [reply .travis.yml]: Fixed on last commit.

**Pull request merged**